### DISCUSSION TITLE
Provide GUI for changing of fonts ingame

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#11728] [Features]
## Version of OpenTTD
1.10.3
## Expected result
Possibility to freely choose the size and font type in the game options window or the settings window, without having to close the game, open the configuration file and also knowledge what and where to change.
## Actual result
The player has very limited options to change the font by step-size multiplier of the font size: x1, x2, x4, which rarely gives a satisfactory result. This choice often doesn't allow to adjust the fonts in the game to your monitor and the player's needs.
## Comment
The default setting is especially disadvantageous for the smallest font. Its enlargement by the available setting makes the average font unnecessarily large.
On the other hand, the default medium font, although quite good in the game, works very badly in a readme file, which is tiring to read after the first line of text.
Here all the work that nielsmh is doing with the manual window (#7786) seems meaningless. :( Even if the entire encyclopedia was posted there, reading it would still be very tedious. Yes, you can change this font, but even many players involved do not know about this possibility, and since the manual will be mainly for players starting their adventure with the game, it is this they are more unlikely to know how to change it.
The font can be freely selected in the configuration file. It is not complicated at all. However, you need to know where and what to do. You also have to exit the game to make a change. Another downside of this form is that you can't see what the effect will be. Not all fonts look good in-game. To find the right one, sometimes even several dozen attempts are needed, which are associated with closing and opening the game each time.
For me, the ability to change the font during the game would not be particularly important, but it would certainly be useful for most players and for the game itself, the interface of which may overwhelm new players with its complexity and not the best, and sometimes poor readability - the default small font is too small for most modern monitors.

**Comment by TrueBrain** [comment]
I edited this ticket to make sure it can be resolved. There was a lot of "while here, can you fix A B C D E F too" stuff in there, which makes it impossible to resolve this issue ever. It now focuses on a single core element. If you feel strongly about any of the other items, please create new tickets for them. You can view this old version back by clicking the `edited by` drop down in the top post.

@LC-Zorg: I asked you a few times now already to stop sneaking in tons of related-but-not-really-the-same things. It might be well intended, but I explained a few times already that is really is not possible for us to process such things, as it makes the issue unresolvable. Please, focus on a single item, so we can easily process them. Please make our life easier, not more difficult. Tnx!

**Comment by heresyh** [comment]
Been playing around a lot with font choices and sizes in the new beta to find something that is pleasing, and the process as is today is quite tedious when you want to do quick testing.

As a middle-ground without going all out and implementing f.ex the font picker patch or such with the work that entails, would it be possible to expose the font settings to the console and have a new command to reload the fonts ingame?

Usage would be f.ex:
set small_font "Comic Sans MS"
reloadfonts

And it would reload the font settings based on what is in openttd.cfg (or possibly fonts.cfg, if it would be easier to implement something by separating the config from the main config file).

**Comment by heresyh** [comment]
Just to reply to myself here, I did a quick test now by adding "_misc_settings," into the "static const SettingTable _generic_setting_tables" and that exposes the font settings (and of course all other settings from that part of the config file).

![image](https://user-images.githubusercontent.com/109810800/209135501-3abf9c39-7be6-47c5-a7b8-9318c693468e.png)

I also commented out line 1152,
//proc(generic_ini, _misc_settings, "misc", nullptr, only_startup);

There are some deviations to when it takes effect, as if I try to do "set medium_size 20" and forcing a reload by changing the language under Game Options, it does not take effect.

However, if I also do 'set medium_font "Tahoma"' (or an invalid font to make it run the fallback routine) and then change the language, both the font size and the font itself takes effect.

I am uncertain what else might break from including "_misc_settings" into the generic table though. Limited testing didn't really show any adverse effects, but this was more a "let me try this quick before I have to return from lunch" thing.
